src/java/com/zimbra/qa/selenium/projects/ajax/tests/mail/group/messages/SortByDateGroupByDateAscending.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2011, 2013, 2014, 2015, 2016 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.qa.selenium.projects.ajax.tests.mail.group.messages;
+
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import com.zimbra.qa.selenium.framework.items.FolderItem;
+import com.zimbra.qa.selenium.framework.items.FolderItem.SystemFolder;
+import com.zimbra.qa.selenium.framework.items.MailItem;
+import com.zimbra.qa.selenium.framework.ui.Action;
+import com.zimbra.qa.selenium.framework.ui.Button;
+import com.zimbra.qa.selenium.framework.util.ConfigProperties;
+import com.zimbra.qa.selenium.framework.util.HarnessException;
+import com.zimbra.qa.selenium.framework.util.OperatingSystem;
+import com.zimbra.qa.selenium.framework.util.SleepUtil;
+import com.zimbra.qa.selenium.framework.util.ZAssert;
+import com.zimbra.qa.selenium.framework.util.ZimbraAccount;
+import com.zimbra.qa.selenium.projects.ajax.core.PrefGroupMailByMessageTest;
+import com.zimbra.qa.selenium.projects.ajax.ui.preferences.PagePreferences.Locators;
+import com.zimbra.qa.selenium.projects.ajax.ui.preferences.TreePreferences.TreeItem;
+
+
+public class SortByDateGroupByDateAscending extends PrefGroupMailByMessageTest {
+
+
+	public SortByDateGroupByDateAscending() {
+		logger.info("New "+ SortByDateGroupByDateAscending.class.getCanonicalName());
+		super.startingPage = app.zPagePreferences;
+		
+		super.startingAccountPreferences.put("zimbraPrefReadingPaneLocation", "bottom");
+		super.startingAccountPreferences.put("zimbraPrefGroupMailBy", "message");
+		super.startingAccountPreferences.put("zimbraPrefMailPollingInterval", "500");
+	}
+
+	@Test( description = "Sort a list of messages by Date in ascending order and Group by Date",
+			groups = { "functional", "L2" })
+	
+	public void SortByDateGroupByDateAscending_01() throws HarnessException {
+
+		if (OperatingSystem.isWindows() == true) {
+
+			//-- File to import
+			final String fileName = "mailOutOfOrder.tgz";
```

Comment:
We are importing a file created after exporting an inbox which had mails delivered on different dates. This was required so that some groups of mail based on the date can be created in the mail list view. It is required to reproduce the issue.